internal/pkg/types/signed_message.go

```diff
@@ -31,13 +32,13 @@ type SignedMessage struct {
 // a signature derived from the serialized `msg` and `msg.From`
 // NOTE: this method can only sign message with From being a public-key type address, not an ID address.
 // We should deprecate this and move to more explicit signing via an address resolver.
-func NewSignedMessage(msg UnsignedMessage, s Signer) (*SignedMessage, error) {
+func NewSignedMessage(ctx context.Context, msg UnsignedMessage, s Signer) (*SignedMessage, error) {
```

Comment:
Note the comment above. I don't want to lean into this interface further.

Reply:
That's fine. This was really a refactoring tool, that can go away now.